libs/sdk-bindings/bindings-android/lib/src/main/kotlin/breez_sdk_notification/ForegroundService.kt

```diff
@@ -0,0 +1,141 @@
+import breez_sdk_notification.job.Job
+import breez_sdk_notification.job.LnurlPayInfoJob
+import breez_sdk_notification.job.LnurlPayInvoiceJob
+import breez_sdk_notification.job.ReceivePaymentJob
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import org.tinylog.kotlin.Logger
+
+open class ForegroundService : SdkForegroundService, Service() {
+    private var breezSDK: BlockingBreezServices? = null
+    val serviceScope = CoroutineScope(Dispatchers.Main.immediate + SupervisorJob())
+
+    companion object {
+        private const val TAG = "ForegroundService"
+    }
+
+    // =========================================================== //
+    //                      SERVICE LIFECYCLE                      //
+    // =========================================================== //
+
+    override fun onBind(intent: Intent): IBinder? {
+        return null
+    }
+
+    /** Stop the service */
+    private val shutdownHandler = Handler(Looper.getMainLooper())
+    private val shutdownRunnable: Runnable = Runnable {
+        Logger.tag(TAG).debug { "Reached scheduled shutdown..." }
+        shutdown()
+    }
+
+    override fun pushbackShutdown() {
+        shutdownHandler.removeCallbacksAndMessages(null)
+        shutdownHandler.postDelayed(shutdownRunnable, Constants.SHUTDOWN_DELAY_MS)
+    }
+
+    override fun shutdown() {
+        Logger.tag(TAG).debug { "Shutting down foreground service" }
+        stopForeground(STOP_FOREGROUND_REMOVE)
+        stopSelf()
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        val intentDetails = "[ intent=$intent, flag=$flags, startId=$startId ]"
+        Logger.tag(TAG).debug { "Start foreground service from intent $intentDetails" }
+
+        // Connect to SDK if source intent has data message with valid payload
+        getConnectRequest()?.let { connectRequest ->
+            getJobFromNotification(intent)?.also { job ->
+                launchSdkConnection(connectRequest, job)
+            } ?: run {
+                Logger.tag(TAG).warn { "Received invalid data message." }
+                shutdown()
+            }
+        } ?: run {
+            Logger.tag(TAG).warn { "Missing ConnectRequest." }
+            shutdown()
+        }
+
+
+        return START_NOT_STICKY
+    }
+
+    open fun getConnectRequest(): ConnectRequest? {
+        return null
+    }
+
+    open fun getNotification(intent: Intent?): Notification? {
+        return null
```

Comment:
Shouldn't we provide a default implementation here? Also what use case did you have in mind for users to override this?

Reply:
I wanted to provide a package/dependency that did not contain any references to Google Services or Firebase. So its the responsibility of the implementer use Google/Firebase if they want and to extract the RemoteMessage from the Intent and return a Notification. 

Thinking about it now, maybe the Notification can extend Parcelable and be put in the intent used to start the foreground service. Then this `getNotification` implementation can be avoided. What do you think?

Reply:
> Thinking about it now, maybe the Notification can extend Parcelable and be put in the intent used to start the foreground service. Then this `getNotification` implementation can be avoided. What do you think?

Yes exactly, that actually decouples totally the notification source from the implementation by defining the required input for the service. I think it is a good idea.